### DISCUSSION TITLE
Simplifying init scriipts nfqws

### DISCUSSION THREAD

**Discussion by NewUse** [bol-van/zapret#613] [Q&A] [answered]
Hi all, as I could see the init system is very complex sh-based with a lot off functions, it take a lot of space and is hard to be fully understand, yes, sure it is done to feet very different startup and base systems it is very universal and cool, but not always  it is needed. I am trying to simplify this system in a very base init script for only nfqws daemon for procd openwrt init system.
The first question is about queue number, is it possible to use the same queue number for all type of traffic: http, https, udp(quic), etc., could nfqws determine protocol and stream themself, or we really need  to use different queues for different strategies?

**Comment by bol-van** [comment, marked as answer]
With latest features only one instance is enough.
Filters can direct traffic to separate profiles

**Reply by NewUse** [reply]
where can I read a little bit more about this feature? and about command line sytax of it?

**Reply by NewUse** [reply]
are you taking about:
--new --filter-l7=http $http_opts --new --filter-l7=tls $https_opts --new --filter-l7=quic $quic_opts
?
wow, it sounds very cool! thanks!!!

**Reply by bol-van** [reply]
Everything is described in readme.txt
--filter-l7 is not a very good choice if you can filter by ip or port
ip/port are known at the beginning, L7 proto - not

**Reply by NewUse** [reply]
thanks again!!
